modules/imgproc/src/color_hsv.simd.hpp

```diff
@@ -56,6 +56,7 @@ struct RGB2HSV_b
         int hr = hrange;
         const int* hdiv_table = hr == 180 ? hdiv_table180 : hdiv_table256;
 
+        cv::AutoLock lock(*mutex);
```

Comment:
This change is not valid as it serializes parallel code.

There is more generic problem related to TSAN. It expects `std::atomic` usage - OpenCV codebase is still not ported to them (atomics is C++11 feature).
Solution should be complete - otherwise problem is not resolved.